test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -52,161 +57,272 @@ class database_operation {
       workload_tracking *tracking)
     {
-        WT_CURSOR *cursor;
         WT_SESSION *session;
         wt_timestamp_t ts;
-        int64_t collection_count, key_count, key_cpt, key_size, value_size;
-        std::string collection_name, cfg, home;
-        key_value_t generated_key, generated_value;
+        std::vector<std::string> collection_names;
+        std::vector<std::thread> threads;
+        int64_t collection_count, key_count, key_size, thread_count, value_size;
+        std::string collection_name;
         bool ts_enabled = timestamp_manager->enabled();
-
-        cursor = nullptr;
-        collection_count = key_count = key_size = value_size = 0;
+        thread_manager tm = thread_manager();
 
         /* Get a session. */
         session = connection_manager::instance().create_session();
-        /* Create n collections as per the configuration and store each collection name. */
-        collection_count = config->get_int(COLLECTION_COUNT);
-        for (size_t i = 0; i < collection_count; ++i) {
-            collection_name = "table:collection" + std::to_string(i);
-            database.add_collection(collection_name);
-            testutil_check(
-              session->create(session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
-            ts = timestamp_manager->get_next_ts();
-            tracking->save_schema_operation(
-              tracking_operation::CREATE_COLLECTION, collection_name, ts);
-        }
-        debug_print(std::to_string(collection_count) + " collections created", DEBUG_TRACE);
 
-        /* Open a cursor on each collection and use the configuration to insert key/value pairs. */
-        key_count = config->get_int(KEY_COUNT);
+        /* Get our configuration values, validating that they make sense. */
+        collection_count = config->get_int(COLLECTION_COUNT);
+        key_count = config->get_int(KEY_COUNT_PER_COLLECTION);
         value_size = config->get_int(VALUE_SIZE);
+        thread_count = config->get_int(THREAD_COUNT);
+        testutil_assert(collection_count % thread_count == 0);
         testutil_assert(value_size > 0);
         key_size = config->get_int(KEY_SIZE);
         testutil_assert(key_size > 0);
+
         /* Keys must be unique. */
         testutil_assert(key_count <= pow(10, key_size));
 
-        for (const auto &collection_name : database.get_collection_names()) {
-            key_cpt = 0;
-            /*
-             * WiredTiger lets you open a cursor on a collection using the same pointer. When a
-             * session is closed, WiredTiger APIs close the cursors too.
-             */
+        /* Create n collections as per the configuration and store each collection name. */
+        for (int64_t i = 0; i < collection_count; ++i) {
+            /* FIXME-T-F: Should we just give collection creation power to the database? */
+            collection_name = database.add_collection();
```

Comment:
Here I am a bit worried that someone might expect add_collection to actually create the collection, let me know what you think.

Reply:
database.add_collection(collection_name) would be more user friendly imo ?

Reply:
We'd need to give each collection a unique Id that maps to a name internally to avoid locking in the database_model when someone would like to know the URI associated with a collection id. Which I don't think would be too ugly?

Reply:
I like the idea.
That's an improvement that can be done in another ticket.